src/components/Comments/TextField.tsx

```diff
@@ -0,0 +1,126 @@
+import {
+  FC, ChangeEvent, useState, useEffect,
+} from 'react';
+import classNames from 'classnames';
+import { FieldIcon, Field } from '../../types';
+
+type Props = {
+  field: Field;
+  type: 'text' | 'textarea';
+  id: string;
+  label: string;
+  placeholder: string;
+  name: 'name' | 'email' | 'body';
+  icon?: FieldIcon;
+  isErrorIcon?: boolean;
+  errorMessage: string;
+  isSubmitted: boolean;
+  isReseted: boolean;
+  value: string;
+  onChange: (value: string) => void;
+  setIsSubmitted: (value: boolean) => void;
+};
+
+export const TextField: FC<Props> = (props) => {
```

Comment:
The component is quite large and doing many things. Consider breaking it down into smaller, more manageable components. This will make your code easier to read, test, and maintain.